pkgs/applications/blockchains/bitcoin.nix

```diff
@@ -1,17 +1,27 @@
-{ stdenv, fetchurl, pkgconfig, autoreconfHook, openssl, db48, boost, zeromq, rapidcheck
-, zlib, miniupnpc, qtbase ? null, qttools ? null, wrapQtAppsHook ? null, utillinux, protobuf, python3, qrencode, libevent
+{ stdenv, fetchFromGitHub, pkgconfig, autoreconfHook
+, openssl, db48, boost, zeromq, rapidcheck, zlib, miniupnpc, utillinux, protobuf
+, python3, qrencode, libevent
+, qtbase ? null, qttools ? null, wrapQtAppsHook ? null
 , withGui }:
 
 with stdenv.lib;
-stdenv.mkDerivation rec{
-  name = "bitcoin" + (toString (optional (!withGui) "d")) + "-" + version;
+
+stdenv.mkDerivation rec {
+  pname = if withGui then "bitcoin" else "bitcoind";
   version = "0.18.1";
 
-  src = fetchurl {
-    urls = [ "https://bitcoincore.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
-             "https://bitcoin.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
-           ];
-    sha256 = "5c7d93f15579e37aa2d1dc79e8f5ac675f59045fceddf604ae0f1550eb03bf96";
+  packaging = fetchFromGitHub {
+    owner = "bitcoin-core";
+    repo = "packaging";
+    rev = "a48094dca1113fb6096768993d1b80d1a4ab5871";
+    sha256 = "0kkbpw4kcxffdik35vvf35vdkcpjfacj1m6vxr73v4f1fzw1kx9m";
+  };
+
+  src = fetchFromGitHub {
```

Comment:
why was this changed to the github repo? I would prefer if we stuck to the tarballs signed with the bitcoin release key.

Reply:
I had to switch source from one on bitcoin.org to github, as distributed package does not include pixmaps. This is also something that archlinux uses.

Reply:
I can also only copy pixmaps from GitHub and build from existing source

Reply:
perhaps it would be better if we added the pixmaps to the packaging repo?

Reply:
or fetch it directly from github ? https://github.com/bitcoin-core/packaging/pull/27/files#diff-5fde7a6d86053f0e1d88c0a2a238941f

Reply:
Agreed, i also made downloads optional only if `withGui` is true, so it should only affect `bitcoin` package and not `bitcoind`